src/ProjectAudioManager.cpp

```diff
@@ -798,14 +798,21 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
 
    bool appendRecord = !sequences.captureSequences.empty();
 
-   auto makeNewClipName = [&](WaveTrack* track) {
+   auto insertEmptyInterval = [&](WaveTrack &track, double t0) {
+      wxString name;
       for (auto i = 1; ; ++i) {
          //i18n-hint a numerical suffix added to distinguish otherwise like-named clips when new record started
-         auto name = XC("%s #%d", "clip name template")
-            .Format(track->GetName(), i).Translation();
-         if (track->FindClipByName(name) == nullptr)
-            return name;
+         name = XC("%s #%d", "clip name template")
+            .Format(track.GetName(), i).Translation();
+         if (track.FindClipByName(name) == nullptr)
+            break;
       }
+
+      auto clip = track.CreateWideClip(t0, name);
+      // So that the empty clip is not skipped for insertion:
+      clip->SetIsPlaceholder(true);
```

Comment:
`InsertInterval` calls `InsertClip` which doesn't do it for empty clips unless they are placeholders.

Reply:
I see, I missed that line.
The emptiness-check before insertion was probably introduced with 1073cef. The commit title says "Fix case of record, stop, put cursor right of clip, record again" ; is this problem specific to your branch ? I suspect it may be on 3.4.2.

Reply:
Background: I'd like to avoid this hacky solution. It's doesn't make sense. An alternative would to pass a flag `bool acceptEmptyClips` defaulting to false. But understanding why the problem is on your branch and not on 3.4.2 first (which I'm not sure of) could yield a better solution.

Reply:
The problem was specific to the branch.

Let me make sufficient conservative transformations now and worry about un-hacking Vitaly's change later.  I want this branch un-blocked.

Reply:
All right.